src/test/kotlin/org/rust/ide/actions/RsFunctionMoverTest.kt

```diff
@@ -0,0 +1,422 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.actions
+
+class RsFunctionMoverTest : RsBaseMoverTest() {
+
+    fun `test step selection`() = moveBothDirectionTest("""
+            // - main.rs
+            fn <selection>test() {</selection>
+
+            }
+
+        """, """
+            // - main.rs
+
+            fn <selection>test() {</selection>
+
+            }
+        """)
+
+    fun `test step`() = moveBothDirectionTest("""
+            // - main.rs
+            fn /*caret*/test() {
+
+            }
+
+        """, """
+            // - main.rs
+
+            fn /*caret*/test() {
+
+            }
+        """)
+
+    fun `test step over function multi liner`() = moveBothDirectionTest("""
+            // - main.rs
+            fn /*caret*/test() {
+
+            }
+            fn foo() {
+
+            }
+        """, """
+            // - main.rs
+            fn foo() {
+
+            }
+            fn /*caret*/test() {
+
+            }
+        """)
+
+    // This test fails in run intellij
```

Comment:
Yeah, that's a complete mystery to me as well!

Unfortunately, I have no better hints then step through in test and in live IDEA and to try to find out, what's different.